### PR TITLE
valueOf method for EnumElement

### PR DESCRIPTION
**Pull request by tjanc**
Adds support for the method introduced in PR #146 on Enum Element.

### CODE REVIEW

File: packages/minim-api-description/lib/define-value-of.js [hunk 1]
```diff
@@ -88,33 +90,45 @@ module.exports = (namespace) => {
...
-      return wrapResult(dflt.content, 'default', opts);
+      return wrapResult(recurse(dflt.content, clearFlag(SOURCE_FLAG, opts)), 'default', opts);
     }
     if (isFlag(NULLABLE_FLAG, opts)) {
       return wrapResult(null, 'nullable', opts);
     }
+    if (isEnumElement(e)) {
+      const enums = e.enumerations;
+      if (enums && enums.content && enums.content[0]) {
+        return wrapResult(recurse(enums.content[0], clearFlag(SOURCE_FLAG, opts)), 'generated', opts);
```
Comment (pksunkara) [inline packages/minim-api-description/lib/define-value-of.js]:
```suggestion
        return wrapResult(recurse(enums.content[0], clearFlag(SOURCE_FLAG, opts)), 'content', opts);
```

File: packages/minim-api-description/test/value-of-test.js [hunk 2]
```diff
@@ -413,3 +414,212 @@ describe('valueOf StringElement with source', () => {
...
+  it('prefers a sample over a default', () => {
+    const element = new EnumElement();
+    element.enumerations = new ArrayElement([
+      new BooleanElement(),
+      new NumberElement(),
+      new StringElement(),
+    ]);
+    element.attributes.set('default', new EnumElement('moin'));
+    element.attributes.set('samples', new ArrayElement([
+      new EnumElement(new EnumElement('zdravicko')),
```
Comment (pksunkara) [inline packages/minim-api-description/test/value-of-test.js]: Why is this double EnumElement?

File: packages/minim-api-description/test/value-of-test.js [hunk 3]
```diff
@@ -413,3 +414,212 @@ describe('valueOf StringElement with source', () => {
...
+  it('prefers content over default and samples', () => {
+    const element = new EnumElement('hello');
+    element.enumerations = new ArrayElement([
+      new BooleanElement(),
+      new NumberElement(),
+      new StringElement(),
+    ]);
+    element.attributes.set('default', new EnumElement('moin'));
+    element.attributes.set('samples', new ArrayElement(
+      new EnumElement(new EnumElement('zdravicko'))
```
Comment (pksunkara) [inline packages/minim-api-description/test/value-of-test.js]: Why is this double EnumElement?

File: packages/minim-api-description/lib/define-value-of.js (hunk 1 above)
Comment (pksunkara) [reply packages/minim-api-description/lib/define-value-of.js]: I am probably wrong on this. Not sure. We might want to forward the value from the enumeration to here, right?

Comment (tjanc) [reply packages/minim-api-description/lib/define-value-of.js]: `generated` is correct here; `enumerations` describes the type and taking the first of those _is_ generating a value.

File: packages/minim-api-description/test/value-of-test.js (hunk 2 above)
Comment (tjanc) [reply packages/minim-api-description/test/value-of-test.js]: To test whether recursion works.

File: packages/minim-api-description/test/value-of-test.js (hunk 3 above)
Comment (tjanc) [reply packages/minim-api-description/test/value-of-test.js]: To test whether recursion works.

**Pull request merged**